src/utils.h

```diff
@@ -5,8 +5,11 @@
  * Misc utilities functions and macros. Internal header.
  */
 
+#define SWAP(a, b) ({ typeof(a) t = a; a = b; b = t; })
```

Comment:
Perhaps wouldn't it be safer to wrap `a` and `b` with parenthesis?

Reply:
In this specific case it shouldn’t matter as both variables are also used as LHS of an assignment. I don’t think there’s a case where the parenthesis would matter *and* it would compile